mkdocs/docs/how-to-release.md

```diff
@@ -48,23 +64,46 @@ deprecation_message(
 )
 ```
 
-## Running a release candidate
+### Update Library Version
+
+Update the version in `pyproject.toml` and `pyiceberg/__init__.py` to match the release version. See [#1276](https://github.com/apache/iceberg-python/pull/1276).
+
+## Publishing a Release Candidate (RC)
+
+### Release Types
+
+#### Major/Minor Release
 
-Make sure that the version is correct in `pyproject.toml` and `pyiceberg/__init__.py`. Correct means that it reflects the version that you want to release.
+* Use the `main` branch for the release.
+* Includes new features, enhancements, and any necessary backward-compatible changes.
+* Examples: `0.8.0`, `0.9.0`, `1.0.0`.
 
-### Setting the tag
+#### Patch Release
 
-Make sure that you're on the right branch, and the latest branch:
+* Use the branch corresponding to the patch version, such as `pyiceberg-0.8.x`.
+* Focuses on critical bug fixes or security patches that maintain backward compatibility.
+* Examples: `0.8.1`, `0.8.2`.
 
-For a Major/Minor release, make sure that you're on `main`, for patch versions the branch corresponding to the version that you want to patch, i.e. `pyiceberg-0.6.x`.
+To create a patch branch from the latest release tag:
 
 ```bash
-git checkout <branch>
-git fetch --all
-git reset --hard apache/<branch>
+# Check out the base branch for the patch version
+git checkout pyiceberg-0.8.x
+
+# Create a new branch for the upcoming patch release
+git checkout -b pyiceberg-0.8.1
```

Comment:
What's the added value of having this branch? Once the release is out, there will be a tag, and I don't think there is any value in keeping the branch unless we want to go for a 0.8.1.1 release